src/modules/credential/token/cli.rs

```diff
@@ -0,0 +1,114 @@
+use std::io::{self, Write};
+
+use better_default::Default;
+use rpassword::read_password;
+use serde_json::{Map, Value};
+
+use crate::{
+    api::{
+        auth::LoginHandler,
+        client::Client,
+        secret::{Secret, SecretAuth},
+        HttpResponse,
+    },
+    errors::RvError,
+    logical::field::FieldTrait,
+    rv_error_string,
+};
+
+#[derive(Default)]
+pub struct TokenCliHandler;
+
+impl LoginHandler for TokenCliHandler {
+    fn auth(&self, client: &Client, data: &Map<String, Value>) -> Result<HttpResponse, RvError> {
+        let lookup = if let Some(lookup_value) = data.get("lookup") {
+            lookup_value.as_bool_ex().ok_or(rv_error_string!("Failed to parse \"lookup\" as boolean"))?
+        } else {
+            true
+        };
+
+        let mut token = data["token"].as_str().unwrap_or("").to_string();
+        if token.is_empty() {
+            let mut writer = io::stdout();
+            write!(writer, "Token (will be hidden): ")?;
+            writer.flush()?;
+            let value = read_password().expect("Failed to read token");
+            writeln!(writer)?;
+            token = value;
+        }
+
+        token = token.trim().to_string();
+        if token.is_empty() {
```

Comment:
You can squash them into one `if` statement.

Reply:
> You can squash them into one `if` statement.

This cannot be done because if the token parameter is empty, it needs to be read from stdin. After reading, it still needs to be checked whether the token is empty again.